test/rules/parameter_assignments.dart

```diff
@@ -45,3 +49,29 @@ void otherBad1(int parameter) { // LINT
   parameter += 3;
   print(parameter);
 }
+
+void actuallyGood(int required, {int optional}) { // OK
+  optional ??= 8;
+}
+
+void actuallyGoodPositional(int required, [int optional]) { // OK
+  optional ??= 8;
+}
+
+void onceAgainBad01(int required, {int optional}) { // LINT
+  optional ??= 8;
+  optional = 42;
+}
+
+void onceAgainBad01Positional(int required, [int optional]) { // LINT
+  optional ??= 8;
+  optional = 42;
+}
+
+void onceAgainBad02(int required, {int optional: 42}) { // LINT
+  optional ??= 8;
+}
+
+void onceAgainBad02Positional(int required, [int optional = 42]) { // LINT
+  optional ??= 8;
+}
```

Comment:
I think the current implementation will not report a lint for the following:
```
void multipleConditionalAssignments({int optional}) {
  optional ??= 4;
  optional ??= 8;
}
```
Should it?